Neos.Fusion/Classes/FusionObjects/MemoImplementation.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Neos\Fusion\FusionObjects;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Memo object that returns the result of previous calls with the same "discriminator"
+ */
+class MemoImplementation extends AbstractFusionObject
+{
+    static protected $cache = [];
+
+    /**
+     * Return the processed value or its cached version based on the discriminator
+     *
+     * @return mixed
+     */
+    public function evaluate()
+    {
+        $discriminator = $this->getDiscriminator();
```

Comment:
Currently we have `entryIdentifier` as cache identifier name. Wouldn't it make sense to use the same name to not make the user remember a new key?

Reply:
I choose it because I had the behaviour of the dynamic cache in the back of my head and I thought about having some differentiation to other caching so people don't expect it to behave exactly the same.

Reply:
+1 on not reusing the same name for different purposes as it could confuse the casual user that this is a different scope.

Reply:
So @daniellienert what do you think?